cluster/extensions/src/main/scala/com/lightbend/lagom/internal/cluster/projections/ProjectorRegistry.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.lightbend.lagom.internal.cluster.projections
+
+import akka.actor.ActorRef
+import akka.actor.ActorSystem
+import akka.actor.Props
+import akka.annotation.ApiMayChange
+import akka.pattern.ask
+import akka.cluster.sharding.ClusterShardingSettings
+import akka.util.Timeout
+import com.lightbend.lagom.internal.cluster.projections.ProjectorRegistry._
+import com.lightbend.lagom.internal.cluster.ClusterDistribution
+import com.lightbend.lagom.internal.cluster.ClusterDistributionSettings
+import com.lightbend.lagom.internal.cluster.projections.ProjectorRegistryActor.DesiredStatus
+import com.lightbend.lagom.internal.cluster.projections.ProjectorRegistryActor.GetStatus
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+@ApiMayChange
+object ProjectorRegistry {
+  sealed trait ProjectorStatus
+  case object Stopped  extends ProjectorStatus
+  case object Starting extends ProjectorStatus
+  case object Running  extends ProjectorStatus
+  case object Stopping extends ProjectorStatus
```

Comment:
because this represents projections/projectors desired states, we should avoid _present continous_ tense. And if we think about `RequestState`, we can say that there are only two: `start` and `stop`.

Reply:
Hmm, my poor naming choices introduced some confusion:

 - this ADT represents the status of a single worker
 - this ADT is currently used in both the desired state (which represents a collection of projectors, its workers and the workers' status), and it is also used in the actual status of a single worker.

I just realized we may need two different models: values for the desired status of a worker and actual status of said worker.

I've locally renamed the trait to `WorkerStatus` to better represent the `WorkerStatus` has 4 possible values and will create an ADT for `RequestedState` which will only have 2 possible values.

Reply:
Something in the lines:

```scala

  sealed trait WorkerStatus
  object WorkerStatus {
    case object Stopped extends WorkerStatus
    case object Stopping extends WorkerStatus
    case object Started extends WorkerStatus
    case object Starting extends WorkerStatus
  }

  sealed trait RequestedState
  object RequestedState {
    case object Stopped extends RequestedState
    case object Started extends RequestedState
  }

```

In any case, this PR will not include the stop/start feature yet so we can postpone the discussion.

Reply:
Fine to postpone that part.

Makes sense to have two states, maybe the exposed states could be: `requestedState` and `observedState`. 

I'm still not convinced that we need `Stopping` and `Starting`. Stopping and starting a stream doesn't take that long. Anyway, let's discuss that when the time comes.